dev/devenv-readme.md

```diff
@@ -12,49 +12,175 @@ First off, if you're on one of the following platforms, we provide the [devenv-i
+    ```
+    # Example incorrect setting
+    $ echo $HOSTTYPE
+    x86_64-cygwin
+
+    # Example correct setting
+    $ echo $HOSTTYPE
+    x86_64
+    ```
+    HOSTTYPE can be overridden on the command line or set in your shell configuration(i.e. .bashrc, .cshrc, etc).  To override HOSTTYPE on the command line use the following construct:
+    ```
+    # for csh/tcsh
+    > setenv HOSTTYPE x86_64 && apt-cyg <command>
+
+    # for bash
+    $ HOSTTYPE=x86_64 apt-cyg <command>
+    ```
 1. Run the script:
 
-    ````
+    ```
     $ ./devenv-install.py
-    ````
+    ```
 
 What does it do for you? Great question! It will create a `~/workspaces/tools/` directory and then download and install (as a user) the following into there for you:
 
 * An [Oracle Java 8 JDK](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html)
-* [Eclipse Mars](https://eclipse.org/mars/)
-    * These Eclipse Plugins:
-        * [m2e-apt](https://developer.jboss.org/en/tools/blog/2012/05/20/annotation-processing-support-in-m2e-or-m2e-apt-100-is-out?_sscc=t): Allows Maven projects in Eclipse to easily leverage [the JDK's Annotation Processing framework](http://docs.oracle.com/javase/7/docs/technotes/guides/apt/).
 * [Apache Maven](https://maven.apache.org/)
+* [Latest Eclipse Version](https://www.eclipse.org/downloads/)
+* Eclipse __m2e-apt__ Plugin
+    * The m2e-apt plugin allows Maven projects in Eclipse to easily leverage [the JDK's Annotation Processing framework](http://docs.oracle.com/javase/7/docs/technotes/guides/apt/).
+    * Install m2e-apt using the [Drag To Install](http://marketplace.eclipse.org/content/m2e-apt) option or install from within the Eclipse IDE by opening **Help > Eclipse Marketplace...** and searching for **m2e-apt** using the **Eclipse Marketplace** find dialog box.  Click the **Install** button for the plugin and restart Eclipse when prompted. 
+    * [Usage Instructions for m2e-apt](https://developer.jboss.org/en/tools/blog/2012/05/20/annotation-processing-support-in-m2e-or-m2e-apt-100-is-out?_sscc=t)
 
 If you're not using one of those supported platforms, or would prefer to setup things manually, you'll want to download and install the items listed above yourself.
 
+## Cygwin Configuration
+
+### Associating .sh files with Cygwin
+Shell scripts that end in .sh should be associated with the Cygwin shell you have chosen to use.  Follow these steps to change the association for your installation:
+
+  1. Open a Command Prompt by pressing the Windows Key+R, type "cmd" and press return.
+  1. In the newly opened command prompt type:
+
+    ``` 
+    # Everything after .sh= is the ftype name
+    > assoc .sh
+    .sh=sh_auto_file
+    ``` 
+  1. Using the ftype name discovered in the previous step check the current association:
+
+    ```
+    # Example of sh_auto_file set to use git-bash
+    > ftype sh_auto_file
```

Comment:
I'd give it a more user-friendly name than `sh_auto_file` -- maybe `cygwin_shell_script`?

Reply:
sh_auto_file seems be be what windows sets it to by default.  That is why I used that name.  I think we might want to keep it because if they do an assoc they will most likely see sh_auto_file and I don't want to confuse them if at all possible.

Reply:
Windows created that association for you?